### ISSUE TITLE
Question about postgresql-cqrs event-modeling view model tables

### ISSUE THREAD

**Issue by artyomsv** [cloudcreate-dk/essentials-spring-examples#29]
Hi, I`m trying to use as a reference project [postgresql-cqrs](https://github.com/cloudcreate-dk/essentials-spring-examples/blob/main/postgresql-cqrs/img/TransferMoneyFlow.png) example but cannot fully understand how green boxes (view models in event-modeling) are being handled. 

To my understanding, we need to store this view data in a specifically prepared db table but from the example, it is not clear when and how to store this data. 

I have tried to store data in my custom created tables inside **@MessageHandler** like:
```java
    @MessageHandler
    void handle(final InterviewSessionStartedEvent event) {
        if (interviewSessions.isInterviewSessionStarted(event.getInterviewSessionId())) {
            interviewSessionRepository.save(new InterviewSessionEntity(event));
        }
    }
```

but that gave me an exception:
```
org.hibernate.PersistentObjectException: detached entity passed to persist: [DB entity classname]
```

Any suggestion on how to handle this scenario?

Thanks in advance!

**Comment by cloudcreate-dk** [comment]
Usually ViewModels are updated by using `EventProcessor`'s (or the new `InTransactionEventProcessor` for in transaction view updates) combined with `MessageHandler` annotated methods. 
Views that use JPA / Spring Data JPA typically work quite well. 
If you use Kotlin, then you can also give the Essentials Postgresql DocumentDB module a try (Note: it's still experimental)

The error seems to indicate that JPA sees the entity as being attached, which can be due to:
- Id generation or value assignment 
- Detached associated entities (other entities attached to the entity you're persisting)

Questions:
- How is the Id annotated field configured?
- How do you handle duplicate Event deliveries?

**Comment by artyomsv** [comment]
Somehow I`m not facing that error any longer. Most probably it was because of the @Id generation. I'm currently using UUID as an ID and generating them when required.

**Comment by artyomsv** [comment]
Your example is using the same processor to handle different events, such as "WithdrawToday" and "DepositToday". In some scenarios, this might be appropriate, but what if I want to isolate slices into separate packages or processors? Do you have any recommendations? So far, I have come to the conclusion that I can use the command bus instance directly in the processor class and emit new commands in the event handler, so they can be picked up by another command handler in another slice processor. Do you think this is the correct approach?

**Comment by cloudcreate-dk** [comment]
Yes, using a single `EventProcessor`, to translate from an Event to a Command, is the best approach for the majority of cases.
Similar to this example:
```
@Component
public class UploadDocumentProcessor extends EventProcessor {
    public UploadDocumentProcessor(EventProcessorDependencies eventProcessorDependencies) {
        super(eventProcessorDependencies);
    }

    @Override
    public String getProcessorName() {
        return "UploadDocumentProcessor";
    }

    @Override
    public List<AggregateType> reactsToEventsRelatedToAggregateTypes() {
        return List.of(Document.AGGREGATE_TYPE);
    }

    @MessageHandler
    void handle(DocumentRegistered e) {
        commandBus.send(
                new UploadDocument(
                        e.getId(),
                        e.getVersion(),
                        e.getAttributes()
                ));
    }
}
```
_Note: It's preferable to use `CommandBus.send` since the `EventProcessor` already guarantees retry handling of the `DocumentRegistered` event, in case the handling of the `UploadDocument` command fails_
